src/Computation/Values/Select.cs

```diff
@@ -47,10 +47,11 @@ public FPExpr AsFloat(IContext context)
 
     public static IValue Create(IValue predicate, IValue trueValue, IValue falseValue)
     {
-        return predicate is IConstantValue p
-            ? p.AsBool()
-                ? trueValue
-                : falseValue
-            : new Select(predicate, trueValue, falseValue);
+        return (predicate, trueValue, falseValue) switch
+        {
+            (IConstantValue p, _, _) => p.AsBool() ? trueValue : falseValue,
+            (_, IConstantValue t, IConstantValue f) when (BigInteger) t.AsUnsigned() == (BigInteger) f.AsUnsigned() => t,
```

Comment:
I think we should do this as

```
(_, IConstantValue t, IConstantValue f) when t.AsUnsigned().Equal(f.AsUnsigned()) => t,
```

using the `Equal` method (not `Equals`) on `ConstantUnsigned` to produce a `ConstantBool` which is then implicitly casted to `bool`. This whole design is maybe not perfect but assuming we are not going to rewrite it all at the moment it is consistent with how everything else works.